### ISSUE TITLE
Conflict with OneSignal on iOS

### ISSUE THREAD

**Issue by luis901101** [MaikuB/flutter_local_notifications#2040]
**Describe the bug**
Notifications doesn't show up in iOS if `onesignal_flutter: ^x.x.x` is one of the project's dependencies.

**To Reproduce**
1. From this plugin example project.
2. Go to `pubspec.yaml`
3. Add `onesignal_flutter: ^3.5.1` to the `dependencies:`
4. Run the example project
5. Tap on **Show plain notification with payload**
6. Notification doesn't show up.

**Expected behavior**
Notification should show up.

**Sample code to reproduce the problem**
Just adding `onesignal_flutter` to the plugin's example project is enough to reproduce the problem.

**Extra info**
I tested it with latest stable `onesignal_flutter: ^3.5.1` and latest beta `onesignal_flutter: ^5.0.0-beta2` and the result is the same. Other thing, if I remove the following from `AppDelegate.swift`
```swift
if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as UNUserNotificationCenterDelegate
    }
```
the notification do show up, but callback is not fired when the user taps on notification.

**Flutter doctor output**
```
[✓] Flutter (Channel stable, 3.10.3, on macOS 13.0 22A380 darwin-arm64, locale en-US)
    • Flutter version 3.10.3 on channel stable at /Users/user/fvm/versions/3.10.3
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f92f44110e (5 weeks ago), 2023-06-01 18:17:33 -0500
    • Engine revision 2a3401c9bb
    • Dart version 3.0.3
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/user/Android/SDK
    • Platform android-33, build-tools 33.0.1
    • ANDROID_HOME = /Users/user/Android/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
```

**Comment by MaikuB** [comment]
I don't use OneSignal and suggest you look on their GitHub repo for a solution as I recalled their team provided a solution https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/197#issuecomment-812660051. A quick search would also show this https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/591#issuecomment-1254983726.

**Comment by luis901101** [comment]
Thanks @MaikuB for the references, as I was using onesignal first I assumed it had something to do with `flutter_local_notifications` confs.
This one worked as expected: [OneSignal/OneSignal-Flutter-SDK#591 (comment)](https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/591#issuecomment-1254983726)

**Comment by FullstackWEB-developer** [comment]
Hi @luis901101 
I got same issue but above solution isn't working on latest versions
#https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/591#issuecomment-2359476439

Could you please check my comment?

**Comment by luis901101** [comment]
> Hi @luis901101 I got same issue but above solution isn't working on latest versions #[OneSignal/OneSignal-Flutter-SDK#591 (comment)](https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/591#issuecomment-2359476439)
> 
> Could you please check [my comment](https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/591#issuecomment-2359476439)?

Hi, checking your `AppDelegate.swift` uncomment this:

```swift
if #available(iOS 10.0, *) {
  UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
}
```

I assume the issue you are having is that you are not able to show local notifications... I hope you can make it work by uncommenting that code in the AppDelegate... anyway regarding OneSignal PushNotifications I think you should handle the `addForegroundWillDisplayListener` so the Push Notifications from OneSignal are treated as local notifications this way you control the way all notifications are shown.

**Comment by FullstackWEB-developer** [comment]
Thanks @luis901101 
It's working well now.
This is a huge help to me.
